src/helpers/createChart.ts

```diff
@@ -31,26 +29,37 @@ module.exports = (data, path) => {
     .attr('transform', `translate(${margin.left}, ${margin.top})`);
 
   // Data mapping
-  const chartData = data.map(d => d.map((val, i) => ({
-    x: i,
-    y: val
-  })));
+  const chartData: { x: number, y: number }[][] = data.map(
+    (d: Array<number>) => d.map(
+      (val: number, i) => ({
+        x: i,
+        y: val,
+      }),
+    ),
+  );
 
   // Get the largest of the two arrays so we can scale the graph properly
-  const largest = chartData.sort((a, b) => a.length < b.length)[0];
+  const largest: { x: number, y: number }[] = chartData.sort(
+    (a: { x: number, y: number }[], b: { x: number, y: number }[]) => a.length - b.length,
+  )[0];
 
   // Dimension functions, used for data scaling
-  const x = d3.scaleLinear().range([0, w]);
-  const y = d3.scaleLinear().range([h, 0]);
+  const x: d3.ScaleLinear<number, number> = d3.scaleLinear().range([0, w]);
+  const y: d3.ScaleLinear<number, number> = d3.scaleLinear().range([h, 0]);
 
   // Use scaling functions from before to create a new line creating function
-  const lineFunc = d3.line()
-    .x(d => x(d.x))
-    .y(d => y(d.y))
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  const lineFunc: Function = d3.line()
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    .x((d: any) => x(d.x))
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    .y((d: any) => y(d.y))
     .curve(d3.curveLinear);
 
-  x.domain(d3.extent(largest, d => d.x));
-  y.domain([0, d3.max(largest, d => d.y)]);
+  // @ts-expect-error
+  x.domain(d3.extent(largest, (d) => d.x));
+  // @ts-expect-error
+  y.domain([0, d3.max(largest, (d) => d.y)]);
```

Comment:
Either add descriptions to the errors or push the types into what they need to be.